lib/charms/tls_certificates_interface/v2/tls_certificates.py

```diff
@@ -1159,6 +1168,84 @@ def _revoke_certificates_for_which_no_csr_exists(self, relation_id: int) -> None
                 )
                 self.remove_certificate(certificate=certificate["certificate"])
 
+    def get_requirer_units_csrs_with_no_certs(
+        self,
+    ) -> List[Dict[str, Union[int, str, List[Dict[str, str]]]]]:
+        """Filters the requirer's units csrs.
+
+        Keeps the ones for which no certificate was provided.
+
+        Returns:
+            list: List of dictonries that contain the unit's csrs
+            that don't have a certificate issued.
+        """
+        all_unit_csr_mappings = copy.deepcopy(self.get_requirer_csrs_by_unit())
+        for unit_csrs in all_unit_csr_mappings:
```

Comment:
`unit_csr_mapping` would probably be a better variable name and avoid us having to call `unit_csrs["unit_csrs"]`

Reply:
Agreed.